src/web-storage.js

```diff
@@ -0,0 +1,137 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+/**
+ * @file WebStorage Wrapper
+ * @description WebStorage Wrapper, uses localStorage when available
+ * Triggers an event if there is a change on a web storage key.
+ * @requires jquery
+ * @requires ju-shared/observable-class
+ * @requires ju-shared/util
+ * @module ju-shared/web-storage
+ * @listens module:ju-shared/web-storage#storageEvent
+ * @fires module:ju-shared/web-storage#storageEvent
+ * @extends ju-shared/observable-class
+ */
+
+define([
+        'jquery',
+        'ju-shared/observable-class',
+        'ju-shared/util'
+    ],
+    function (
+        $,
+        ObservableClass,
+        Util
+    ){
+
+        'use strict';
+
+        var WebStorage = ObservableClass.extend({
+
+            /**
+             * @constructor
+             * @alias module:ju-shared/web-storage
+             * @param {Object} [opts] - configuration options
+             * @param {Boolean} [opts.fireStorageEvent=true] - activate the listener of storage event, it is fired when a storage area (localStorage or sessionStorage) has been modified.
+             */
+            init : function (opts) {
+                this.fireStorageEvent = true;
+                this.isLocalStorageAvailable = this.getIsLocalStorageAvailable();
+
+                if (opts) {
+                    this.fireStorageEvent = opts.fireStorageEvent || this.fireStorageEvent;
+                }
+
+                if (this.isLocalStorageAvailable && this.fireStorageEvent){
```

Comment:
I prefer this to be moved to a separate method.

That way we don't have side effects and we won't need a flag for `fireStorageEvent` as we could just say: `var storage = new WebStorage(); storage.listenStorageEvents();` and if we don't want such event, we just don't call the second instruction